File: IT/Database/Esercizi/Java/Esercizio12.java
/*
file Esercizio12.java
eseguire con: java -cp .:sqlitejdbc-v<nnn>.jar Esercizio12
*/

import java.sql.DriverManager;
import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.Statement;
import java.sql.SQLException;

public class Esercizio12 {
	public static void main(String [] args) {
		Connection conn = null;
		Statement stmt = null;
		ResultSet res = null;

		try {
			Class.forName("org.sqlite.JDBC");
			String dbName = "students.db";
			conn = DriverManager.getConnection("jdbc:sqlite:" + dbName);
			stmt = conn.createStatement();

			res = stmt.executeQuery("SELECT MIN(grade) FROM students");

			int min = 0;

			if (res.next()) {
				min = res.getInt(1);
			}

			if (min == 0) {
				System.out.println("Nessuno studente ha ancora passato l'esame.");
				res.close();
				stmt.close();
				conn.close();
				return;
			}

			System.out.println("Voto minimo: " + min);
		}
		catch (ClassNotFoundException e) {
			System.err.println(e);
		}
		catch (SQLException e) {
			System.err.println(e);
		}
		finally {
			try {
				if (res != null) res.close();
				if (stmt != null) stmt.close();
				if (conn != null) conn.close();
			}
			catch (SQLException e) {
				System.err.println(e);
			}
		}
	}
}

File: IT/2/Database/Esercizi/C++/esercizio12.cpp
/* file esercizio12.cpp */

#include <iostream>
#include <sqlite3.h>

using std::cout;
using std::endl;

int main()
{
	sqlite3 *db;
	sqlite3_stmt *res;

	sqlite3_open("students.db", &db);
	int rc = sqlite3_prepare_v2(db, "SELECT MIN(grade) FROM students", -1, &res, NULL);

	int min = 0;

	if (rc != SQLITE_ERROR) {
		if (sqlite3_step(res) == SQLITE_ROW) {
			min = sqlite3_column_int(res, 0);
		}

		if (!min) {
			cout << "Nessuno studente ha ancora passato l'esame." << endl;
			sqlite3_finalize(res);
			sqlite3_close(db);
			return 0;
		}

		cout << "Voto minimo: " << min << endl;
	}
	else {
		cout << sqlite3_errmsg(db) << endl;
	}

	sqlite3_finalize(res);
	sqlite3_close(db);
}

